openedx/core/djangoapps/content/search/plain_text_math.py

```diff
@@ -0,0 +1,105 @@
+"""
+Helper class to convert mathjax equations to plain text.
+"""
+
+import re
+
+import unicodeit
+
+
+class PlainTextMath:
+    """
+    Converts mathjax equations to plain text using unicodeit and some preprocessing.
+    """
+    equation_pattern = re.compile(
+        r'\[mathjaxinline\](.*?)\[\/mathjaxinline\]|\[mathjax\](.*?)\[\/mathjax\]|\\\((.*?)\\\)|\\\[(.*?)\\\]'
+    )
+    eqn_replacements = (
+        # just remove prefix `\`
+        ("\\sin", "sin"),
+        ("\\cos", "cos"),
+        ("\\tan", "tan"),
+        ("\\arcsin", "arcsin"),
+        ("\\arccos", "arccos"),
+        ("\\arctan", "arctan"),
+        ("\\cot", "cot"),
+        ("\\sec", "sec"),
+        ("\\csc", "csc"),
+        # Is used for matching brackets in mathjax, should not be required in plain text.
+        ("\\left", ""),
+        ("\\right", ""),
+    )
+    regex_replacements = (
+        # Makes text bold, so not required in plain text.
+        (re.compile(r'\\mathbf{(.*?)}'), r"\1"),
+        (re.compile(r'{\\bf (.*?)}'), r"\1"),
+    )
+
+    def _fraction_handler(self, equation: str) -> str:
+        """
+        Converts `\\frac{x}{y}` to `(x/y)` while handling nested `{}`.
```

Comment:
```suggestion
        Converts `\frac{x}{y}` to `(x/y)` while handling nested `{}`.
```
In the docs and anything that's not a python string, it's best to represent these as users enter them, with just a single `\`.

Reply:
Pylint complains when we have single `\` in docstrings. There is an [issue](https://github.com/pylint-dev/pylint/issues/264) with a lot of discussion about the same which suggests using `r"""` in such cases.

Reply:
Ah, I see. Well it's fine with me either way; I'll leave it to your judgement.

Reply:
Used `r"""` for docs containing these examples.